src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -777,26 +804,51 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     if (runnableCalls.nonEmpty)
       logger.info(s"starting calls: " + runnableCalls.map(_.fullyQualifiedName).toSeq.sorted.mkString(", "))
 
-    val entries: Traversable[Try[ExecutionStartResult]] = runnableEntries map {
-      case (k: ScatterKey, _) => processRunnableScatter(k)
-      case (k: CollectorKey, _) => processRunnableCollector(k)
-      case (k: BackendCallKey, _) => processRunnableCall(k)
-      case (k: FinalCallKey, _) => processRunnableCall(k)
+    val (callKeyedRunnableEntries, unexpectedRunnableEntries)= runnableEntries.partition {
+      case (k: CallKey, _) => true
+      case _ => false
+    }
+
+    val callKeyEntries: Map[CallKey, Try[ExecutionStartResult]] = callKeyedRunnableEntries collect {
+      case (k: BackendCallKey, _) => k -> processRunnableCall(k)
+      case (k: FinalCallKey, _) => k -> processRunnableCall(k)
+      case (k: CallKey, v) =>
+        val message = s"Unknown entry in execution store:\nKEY: $k\nVALUE:$v"
+        logger.error(message)
+        k -> Failure(new UnsupportedOperationException(message))
+    }
+
+    val nonCallKeyEntries = unexpectedRunnableEntries map {
+      case (k: ScatterKey, _) => k -> processRunnableScatter(k)
+      case (k: CollectorKey, _) => k -> processRunnableCollector(k)
       case (k, v) =>
         val message = s"Unknown entry in execution store:\nKEY: $k\nVALUE:$v"
         logger.error(message)
-        Failure(new UnsupportedOperationException(message))
+        k -> Failure(new UnsupportedOperationException(message))
+    }
+
+    // Handle call key failures:
+    callKeyEntries filter (_._2.isFailure) foreach { case (key: CallKey, failure) =>
+      self ! CallFailedToInitialize(key)
     }
 
-    entries.find(_.isFailure) match {
-      case Some(failure) => failure
-      case _ =>
-        // Coalesce all the ExecutionStartResults into one all-encompassing ExecutionStartResult.
-        val zero = ExecutionStartResult(startEntries = Set.empty)
-        Success(entries.foldLeft(zero) { case (acc, e) => acc plus e.get } )
+    // Handle non-call key failures:
+    nonCallKeyEntries foreach {
+      case (_, Failure(e)) =>
+        logger.error(e.getMessage, e)
+        self ! CheckForWorkflowComplete
```

Comment:
and why is the handling of completely unrecognized keys so different from unrecognized call keys?